Source/IO.cpp

```diff
@@ -1194,3 +1195,102 @@ PeleC::writeSmallPlotFile(
   TheFullPath += BaseName;
   amrex::VisMF::Write(plotMF, TheFullPath, how, true);
 }
+
+void
+PeleC::initLevelDataFromPlt(
+  const int lev, const std::string& dataPltFile, amrex::MultiFab& state)
+{
+  amrex::Print() << "Using data (rho, u, T, Y) from pltfile " << dataPltFile
+                 << std::endl;
+  pele::physics::pltfilemanager::PltFileManager pltData(dataPltFile);
+  const auto plt_vars = pltData.getVariableList();
+
+  // Read rho, u, temperature (required)
+  std::map<std::string, int> vars{
+    {"density", -1}, {"x_velocity", -1}, {"Temp", -1}};
+  for (auto& var : vars) {
+    var.second = find_position(plt_vars, var.first);
+    if (var.second == -1) {
+      amrex::Abort("Unable to find variable in plot file: " + var.first);
+    }
+  }
+  pltData.fillPatchFromPlt(lev, geom, vars["density"], URHO, 1, state);
+  pltData.fillPatchFromPlt(
+    lev, geom, vars["x_velocity"], UMX, AMREX_SPACEDIM, state);
+  pltData.fillPatchFromPlt(lev, geom, vars["Temp"], UTEMP, 1, state);
+
+  // Copy species from the plot file
+  amrex::Vector<std::string> spec_names;
+  pele::physics::eos::speciesNames<pele::physics::PhysicsType::eos_type>(
+    spec_names);
+  for (int n = 0; n < spec_names.size(); n++) {
+    const auto& spec = spec_names.at(n);
+    const int pos = find_position(plt_vars, "Y(" + spec + ")");
+    if (pos != -1) {
+      pltData.fillPatchFromPlt(lev, geom, pos, UFS + n, 1, state);
+    }
+  }
+
+  // Sanity check the species, clean them up if they aren't too bad
+  auto sarrs = state.arrays();
+  amrex::ParallelFor(
+    state, [=] AMREX_GPU_DEVICE(int nbx, int i, int j, int k) noexcept {
+      auto sarr = sarrs[nbx];
+      amrex::Real sumY = 0.0;
+      const auto tol = constants::small_num();
+
+      for (int n = 0; n < NUM_SPECIES; n++) {
+        // if the species is not too far out of bounds, clip it
+        const auto mf = sarr(i, j, k, UFS + n);
+        if ((mf < 0.0) || (1.0 < mf)) {
+          if (((-tol < mf) && (mf < 0.0)) || ((1.0 < mf) && (mf < 1 + tol))) {
```

Comment:
How about making `tol` something that is just defined for this function, or even a parmparsing it? For the use case of going between skeletal and reduced mechanisms, I could see it being useful to control this independently of whetever else `constants::small_num()` controls. I agree with this comment from @drummerdoc https://github.com/AMReX-Combustion/PeleC/pull/426#issuecomment-1064641187

Reply:
Good idea. I added that too.